### PR TITLE
[T-6543][17.0][MIG+IMP] queued_sale_notify: migration + refactor

### PR DESCRIPTION
**Pull request by Tisho99**

### CODE REVIEW

**Comment by Tisho99** [comment]
@ValentinVinagre @HaraldPanten @luis-ron 

The PR is ready

**Comment by HaraldPanten** [comment]
Don't you need to check tests?

**Comment by Tisho99** [comment]
@HaraldPanten 

No, that check if failing because i added a test-requierments.txt file in a "DON'T MERGE" commit with the dependencies of other PR. When the base module is merged, i will remove that commit and everything will be ocay

**Comment by HaraldPanten** [comment]
@Tisho99 OK then. Let's wait for other reviewers

**Review by luis-ron — APPROVED** [review]
Functional review: LGTM 👍🏻

File: queued_sale_notify/models/sale_notify_mixin.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    )
+
+    sale_order_type_id = fields.Many2one(
+        name="Sale Order Type", comodel_name="sale.order.type"
+    )
+
+    # TODO: Mover a private o borrarlo
+    def is_to_notify(self, record):
+        # Returns if a record is in a status to be notified
+        return self.trigger_state == "sale" and record.state == "sale"
```
Comment (manuelregidor) [inline queued_sale_notify/models/sale_notify_mixin.py:33]: Shouldn't done state be considered?

File: queued_sale_notify/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        res = super().action_confirm()
+        notity_set = self.type_id._get_sale_notify_ids()
+        for notify_ids in notity_set.values():
+            for notify_id in notify_ids:
+                for rec in self:
+                    if notify_id.is_to_notify(rec):
```
Comment (manuelregidor) [inline queued_sale_notify/models/sale_order.py:16]: Couldn't you get rid of this check? It checked again when notify is called.

**Review by manuelregidor — CHANGES_REQUESTED** [review]
Also, either credit me or remove my name from files.

File: queued_sale_notify/models/sale_notify_mixin.py (hunk 1 above)
Comment (Tisho99) [reply queued_sale_notify/models/sale_notify_mixin.py:33]: There is no done state in v17

**Comment by manuelregidor** [comment]
Technical review. LGTM

**Review by manuelregidor — APPROVED** [review]

**Pull request merged**